tests/integration/test_osm_mysql.py

```diff
@@ -0,0 +1,161 @@
+                "keystone-image=opensourcemano/keystone:testing-daily",
+                "osm-keystone",
+                "osm-keystone",
+            ),
+            ops_test.model.deploy(
+                "osm-pol",
+                application_name="osm-pol",
+                channel="latest/candidate",
+                resources=osm_pol_resources,
+            ),
+            ops_test.model.deploy("charmed-osm-kafka-k8s", application_name="osm-kafka"),
+            ops_test.model.deploy("charmed-osm-zookeeper-k8s", application_name="osm-zookeeper"),
+            ops_test.model.deploy("charmed-osm-mongodb-k8s", application_name="osm-mongodb"),
+        )
+
+        # cannot block until "osm-keystone" units are available since they are not
+        # registered with ops_test.model.applications (due to the way it's deployed)
+        await ops_test.model.block_until(
+            lambda: len(ops_test.model.applications[APP_NAME].units) == 3
+        )
+        await ops_test.model.block_until(
+            lambda: len(ops_test.model.applications["osm-pol"].units) == 1
+        )
+        await ops_test.model.block_until(
+            lambda: len(ops_test.model.applications["osm-kafka"].units) == 1
+        )
+        await ops_test.model.block_until(
+            lambda: len(ops_test.model.applications["osm-zookeeper"].units) == 1
+        )
+        await ops_test.model.block_until(
+            lambda: len(ops_test.model.applications["osm-mongodb"].units) == 1
+        )
+
+        await ops_test.model.relate("osm-kafka:zookeeper", "osm-zookeeper:zookeeper")
+        await ops_test.model.block_until(
+            lambda: is_relation_joined(ops_test, "zookeeper", "zookeeper")
+        )
+
+        await ops_test.model.wait_for_idle(
+            apps=[APP_NAME, "osm-kafka", "osm-zookeeper", "osm-mongodb"],
+            status="active",
+            raise_on_blocked=True,
+            timeout=1000,
+        )
+
+        await ops_test.model.relate("osm-keystone:db", f"{APP_NAME}:osm-mysql")
+        await ops_test.model.block_until(lambda: is_relation_joined(ops_test, "db", "osm-mysql"))
+
+        # osm-keystone is initially in blocked status
+        await ops_test.model.wait_for_idle(
+            apps=[APP_NAME, "osm-keystone", "osm-kafka", "osm-zookeeper", "osm-mongodb"],
+            status="active",
+            raise_on_blocked=False,
+            timeout=1000,
+        )
+
+        await ops_test.model.relate("osm-pol:mongodb", "osm-mongodb:mongo")
+        await ops_test.model.block_until(lambda: is_relation_joined(ops_test, "mongodb", "mongo"))
+
+        await ops_test.model.relate("osm-pol:kafka", "osm-kafka:kafka")
+        await ops_test.model.block_until(lambda: is_relation_joined(ops_test, "kafka", "kafka"))
+
+        await ops_test.model.relate("osm-pol:mysql", f"{APP_NAME}:osm-mysql")
+        await ops_test.model.block_until(
+            lambda: is_relation_joined(ops_test, "mysql", "osm-mysql")
+        )
+
```

Comment:
I think it would be nice to break this into two tests

`test_deploy_and_relate` and `test_osm_pol_ops`

Reply:
addressed in 2f6e6e434f2bab7f2a508153eb234fb73edf2bed